### PR TITLE
Add notification button components

### LINKED ISSUE #298: Notification Dropdown
## Description

Add the outer portion of the Notifications Dropdown.

Context: This is where users can see a list of their notifications. 

## Todo

- [x] Add Bell Icon
- [x] Add dropdown
- [x] Add buttons for All and Unread
- [x] Divide dropdown into New and Earlier
- [x] Adapt for both mobile and web

## Useful links

- Mobile Figma Design: https://www.figma.com/file/rdoLGoH5vfFBfqV11lgOyw/ZubHub-Mobile?node-id=0%3A1
- Web Figma: https://www.figma.com/file/jRQdwIwXdgmXEe5L0XLgno/ZubHub-Web?node-id=0%3A1

### PR DESCRIPTION
**Pull request by AndrewLester**
## Summary

Closes #298 

This won't be merged into master without the content being done, perhaps that will be merged in here.

## Changes

- Adds the notification dropdown panel

## Screenshots

In navbar:
image

Opened:
image

image

image

### CODE REVIEW

**Comment by NdibeRaymond** [comment]
Hello @AndrewLester , thanks for working on this feature, the UI looks great!.
- I was wondering, how the notification UX flow look like?
- Is there a little mockup/sketch of this in full ? I saw some figma files but they are all locked (the sketch will probably include the notification objects themselves, menus for opening where you see things like "mark notification as read", "delete notification", etc).
- I think a mockup will really help in iterating fast (and I speak as a frontend person here. somethings it's very painful 
  reimplementing a design just because someone thinks there are some minor things that needs to be fixed. a mockup helps us 
 avoid that pain).
- Also, how will stuffs like pagination be handled ? (since the data returning from the backend will be paginated)

**Comment by AndrewLester** [comment]
@NdibeRaymond , a lot of your questions are probably answered by the figma designs you saw, but I'll answer the last one. We were planning on only displaying the last 10 or so notifications in the panel, so the endpoint would either return only the last 10 or the panel would only need the first page (of size 10). No pagination UI would be available in this panel.

File: zubhub_frontend/zubhub/src/components/notification_panel/NotificationPanel.jsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        {earlierNotificationsLength > 0 && (
+          <h2 className={classes.panelSubheadingTextStyle}>Earlier</h2>
+        )}
+      </div>
+    );
+  };
+
+  const getUnreadNotificationView = () => (
+    <div>Unread notifications here...</div>
+  );
```
Comment (ghost) [inline zubhub_frontend/zubhub/src/components/notification_panel/NotificationPanel.jsx]: Small thing but this doesn't show up under the Unread tab for the desktop view. It does show up under mobile though.

**Review by ghost — APPROVED** [review]
Great work on the PR! There's a small bug on the unread tab but I think that should be quick to fix.

**Pull request merged**